### PR TITLE
Add ability to edit and delete scenarios and measures

### PR DESCRIPTION
**Pull request by perryr16**
#### Any background context you want to provide?
This PR is a precursor to the BPS event tracking. The current state of Scenarios and PropertyMeasures are read-only from a user perspective and can only be generated through a buildingsync or AT upload. There are often non-relavent Scenarios and PropertyMeasures created and displayed with no way of editing or removing them.

#### What's this PR do?
This adds `list`, `retrieve`, `update`, and `destroy` endpoints for Scenarios and PropertyMeasures (a new SEEDOrg viewset has been generated to accomplish this).  This functionality is only available to the backend as the frontend layout needs further design. 

#### How should this be manually tested?
Run the tests, use swagger

#### What are the relevant tickets?
 #3771, #3772,  #3777, #3795

#### Screenshots (if appropriate)
New endpoints in swagger documentation
Screenshot 2023-02-01 at 11 40 03 AM

### CODE REVIEW

**Comment by nllong** [comment]
Hey @perryr16  -- thanks for getting this in there. Just a quick heads up since you probably didn't know of the process update. We are now auto-generating the changelogs from the pull requests whereas before we were creating it from closed issues. So the pull requests title now becomes the change log entry. Can you update this title name to start with a verb and capitalized, e.g., Add ability to delete measures... or whatever.

**Review by nllong — CHANGES_REQUESTED** [review]

File: seed/api/v3/urls.py [hunk 1]
```diff
@@ -110,6 +111,9 @@
 properties_router.register(r'notes', NoteViewSet, basename='property-notes')
 properties_router.register(r'scenarios', PropertyScenarioViewSet, basename='property-scenarios')
 
+property_measures_router = nested_routers.NestedSimpleRouter(properties_router, r'scenarios', lookup='scenario')
+property_measures_router.register(r'property_measures', PropertyMeasureViewSet, basename='property-measures')
```
[the remark this answers is not in the document]
Comment (perryr16) [reply seed/api/v3/urls.py:115]: Can do. This feels related to how the `property` in URL `/properties/101` is actually a `property_view`

[the remark this answers is not in the document]
Comment (nllong) [reply seed/api/v3/urls.py]: exactly. I just want the routes to be reflective of how a building analyst understands the world.

**Review by anchapin — CHANGES_REQUESTED** [review]
This is looking pretty good. However, I have two requests:

1. I expected that the "id" returned with "v3_properties_scenarios_measures_list" would be the id from the "propertymeasure" table in the database not the one from the "measure" table. This appears to be coming from the "PropertyMeasureSerializer" [here](https://github.com/SEED-platform/seed/blob/40ee2e609a7e4d94205b0c7afb6a08f6fb1b51ab/seed/serializers/measures.py#L21). This code was edited last by @nllong so can you chime in on whether this was done intentionally or not?
2. The data types listed in swagger for the "v3_properties_scenarios_measures_update" endpoint don't match what is actually expected. For example, "category_affected" is listed as data type of string but is actually expected to be an int.

**Comment by perryr16** [comment]
@anchapin regarding 
>The data types listed in swagger for the "v3_properties_scenarios_measures_update" endpoint don't match what is actually expected. For example, "category_affected" is listed as data type of string but is actually expected to be an int.

I agree. It's a bit confusing. However, these fields are enum fields that take an integer and return a string. I believe the desired pattern is to input `"category_affected": 1` and return `"category_affected": "Heating System"`. I can change the field to accept an integer and return an integer but that might contradict the intended functionality. 

thoughts?

**Comment by perryr16** [comment]
@anchapin, thanks for the feedback. I've manually changed the data types in the request body for the Scenario and Measure update endpoints.
Screenshot 2023-02-06 at 10 23 36 AM
Screenshot 2023-02-06 at 10 23 22 AM

**Comment by anchapin** [comment]
> I expected that the "id" returned with "v3_properties_scenarios_measures_list" would be the id from the "propertymeasure" table in the database not the one from the "measure" table. This appears to be coming from the "PropertyMeasureSerializer" [here](https://github.com/SEED-platform/seed/blob/40ee2e609a7e4d94205b0c7afb6a08f6fb1b51ab/seed/serializers/measures.py#L21). This code was edited last by @nllong so can you chime in on whether this was done intentionally or not?

@nllong can you respond to this question regarding the "PropertyMeasureSerializer"? Let me know if you want to discuss it.

**Comment by nllong** [comment]
Strange, I thought for sure that I responded to the `PropertyMeasureSerializer` question. Regardless, the `de9b4f8` commit above implemented the correct fix... it should have been the `propertymeasure_id`.

**Review by nllong — APPROVED** [review]
measure editing! Looks good.

**Review by anchapin — APPROVED** [review]

**Comment by axelstudios** [comment]
@perryr16 I added some fixes:
- Fixed 500 errors when the input datatypes did not match the expected model types
- Fixed 500 error when using PUT with a nonexistent scenario
- Removed 3 fields from the scenario input model that always resulted in 500 errors: `measures`, `property_state`, and `reference_case`
- Removed unused pagination inputs on measures endpoints
- Fixed listing measures for a scenario, if the scenario exists with zero measures it now returns an empty array rather than a 404
- Fixed measure input model, `recommended` is a boolean rather than a string
- Updated schema generation to allow OpenAPI enum types and document the enum options
- Added enum validation to scenario/measure endpoints. Strings and ints are both accepted and validated
  - Before, this was resulting in 500 errors if invalid options were sent, like `false` or `0` or other invalid enum integers.  Subsequently, any endpoint that pulled in another model with an invalid integer would start returning 500 errors.
- Fixed Inventory Detail duplicate BuildingSync filetype errors (unrelated to this PR)

#### Potential issues that remain unfixed:
- The PUT requests actually behave like PATCH instead, merging partial input data with the existing record
- A measure can be update/deleted from a scenario, but can't be added

**Enum types**:

**Review by axelstudios — APPROVED** [review]

**Comment by perryr16** [comment]
@axelstudios Amazing, thank you for all that work

**Pull request merged**